python/vowpalwabbit/DFtoVWtoTensorboard.py

```diff
@@ -0,0 +1,182 @@
+		weighted_examples_since_last : float
+										This is the difference between current iteration's weighted_examples and previous iteration's weighted_examples which is used to calculate since_last
+
+		Returns
+		-------
+
+		since_last      : float
+							since_last calculated using sum_loss_since_last and weighted_examples_since_last or if ZeroDivisionError is encountered then 0. is returned 
+		""" 
+		try:
+			return sum_loss_since_last / weighted_examples_since_last
+
+		except ZeroDivisionError:
+			return 0.
+
+
+	#------------------------------------------------------------------------------------------------		
+	def _print_metrics(self, average_loss, since_last, label, prediction, num_features):
+		"""This method prints metrics
+
+		Parameters
+		----------
+
+		average_loss : float
+						This is the average_loss value
+		since_last   : float
+						This is the since_last value
+		label        : integer
+						This is the actual label of current example
+		prediction   : integer
+						This is the predicted label of current example
+		num_features : integer
+						This is the num_features value
+
+		Returns
+		-------
+
+		None
+		"""
+		print( 'average_loss:{:.6f}'.format(average_loss) , end='\t')
+		print('since_last:{:.6f}'.format(since_last), end='\t')
+		print('label:', label, end='\t')    
+		print('prediction:', prediction, end='\t')    
+		print('num_features:', num_features) 
+
+
+	#------------------------------------------------------------------------------------------------		
+	def fit(self, vw_to_tensorboard=None):
+		"""Learns on the relevant examples and can also log metrics for tensorboard visualization
+
+		Parameters
+		----------
+
+		vw_to_tensorboard  : VWtoTensorboard object 
+								Default value is None, this parameter is used to control the logging of metrics for Tensorboard visualization
+								If value is VWtoTensorboard object : metrics are logged for Tensorboard visualization
+								If value is None : metrics are not logged for Tensorboard visualization
+
+		Returns
+		-------
+
+		None
+		"""
+		if isinstance(vw_to_tensorboard, VWtoTensorboard):
+			file_writer = tx.SummaryWriter(vw_to_tensorboard.logdir)   # creating file writer
+		
+		sum_loss = 0.
+		weighted_examples = 0.
+
+		for iteration, vw_format in enumerate(self.vw_formatted_data):
+			example = self.vw.parse(vw_format)       # parse the string format, it returns an example object
+			self.vw.learn(example)                  # learn on example
+
+			label = pyvw.get_label(example, self.vw.get_label_type())
+			prediction = pyvw.get_prediction(example, self.vw.get_prediction_type())
+			num_features = example.get_feature_number()    
+
+			self.vw.finish_example(example)  # Any use of vw object should be done after this and use of example before this 
+
+			sum_loss_since_last = self.vw.get_sum_loss() - sum_loss  # vw.get_sum_loss() return current sum loss, sum_loss variable right now holds sum loss of previous iteration
```

Comment:
[Minor][Perf] You should consider making metrics calculation/printing here be optional (similar to how you made the tensorboard output optional.